lib/rules/multiline-ternary.js

```diff
@@ -69,6 +71,10 @@ module.exports = {
                         reportError(node.consequent, node, false);
                     }
                 } else {
+                    if (allowSingleLine && astUtils.isTokenOnSameLine(node, node)) {
```

Comment:
Not to ruin the fun, but I feel like this makes the code a bit hard to read -- semantically a node would always be on the same line as itself, and this only works due to an implementation detail of `astUtils.isTokenOnSameLine` where it assumed that the two tokens would be different. Personally I think it would be better to be explicit here instead:

```js
if (allowSingleLine && node.loc.start.line === node.loc.end.line) {
```